### PR TITLE
NFS Dev Preview : Allow volume expansion for NFS & deploy a NFS volumesnapshot class

### PR DESCRIPTION
**Pull request by malayparida2000**
Refers- [RHSTOR-3435](https://issues.redhat.com/browse/RHSTOR-3435)

For the NFS Dev preview, we are enabling volume expansion for NFS storageclass & deploying a default NFS volume snapshot class.

### CODE REVIEW

**Comment by Rakshith-R** [comment]
/hold 

Still needs changes in rook,
@malayparida2000 this needs to be tested after rook has the changes.
Both upgrade and nfs pvc expansion/snap creation/ restoring to pvc.

**Comment by malayparida2000** [comment]
Yeah, @Rakshith-R , lets keep this on hold till the changes are done on rook side. No hurries.

**Review by nixpanic — COMMENTED** [review]
It would be nice to see an example of the newly generated class. Can that be shared?

File: controllers/storagecluster/generate.go
Comment (Rakshith-R) [inline controllers/storagecluster/generate.go:84]:
```suggestion
	// nfs uses the same cephfs secrets
```

File: controllers/storagecluster/storageclasses.go
Comment (Rakshith-R) [inline controllers/storagecluster/storageclasses.go:255]: Have you checked upgrade ? 

is this field mutable or whether ocs-op deletes and recreates sc  ?

**Review by Rakshith-R — CHANGES_REQUESTED** [review]
minor comments 

We can merged this after they are addressed and rook pr https://github.com/rook/rook/pull/10892
is merged, backported and DS synced.

/cc @BlaineEXE

File: controllers/storagecluster/generate.go
Comment (malayparida2000) [reply controllers/storagecluster/generate.go]: done

File: controllers/storagecluster/storageclasses.go
Comment (malayparida2000) [reply controllers/storagecluster/storageclasses.go:320]: Have to check on this

Comment (Rakshith-R) [reply controllers/storagecluster/storageclasses.go:320]: checked, its mutable

**Review by Rakshith-R — APPROVED** [review]

**Comment by Rakshith-R** [comment]
> It would be nice to see an example of the newly generated class. Can that be shared?

@malayparida2000 ^

**Comment by malayparida2000** [comment]
> > It would be nice to see an example of the newly generated class. Can that be shared?
> 
> @malayparida2000 ^

Are the required PRs now merged? And which version of rook or ceph-csi we need for testing?

**Comment by malayparida2000** [comment]
The required PRs are now merged,
I have completely tested the changes and noted down the changes in this [doc](https://hackmd.io/@Yh4a4hAATcW2BNYBJVSx4w/Skorrt31i)

@nixpanic 
Here is how the newly created volumesnapshotclass looks
`oc get volumesnapshotclass | grep nfs
ocs-storagecluster-nfsplugin-snapclass      openshift-storage.nfs.csi.ceph.com      Delete           5m27s

oc get volumesnapshotclass ocs-storagecluster-nfsplugin-snapclass -o yaml
apiVersion: snapshot.storage.k8s.io/v1
deletionPolicy: Delete
driver: openshift-storage.nfs.csi.ceph.com
kind: VolumeSnapshotClass
metadata:
  creationTimestamp: "2022-09-09T21:03:58Z"
  generation: 1
  name: ocs-storagecluster-nfsplugin-snapclass
  resourceVersion: "304091"
  uid: fef04841-3504-4b30-888e-c4f3d8adc595
parameters:
  clusterID: openshift-storage
  csi.storage.k8s.io/snapshotter-secret-name: rook-csi-cephfs-provisioner
  csi.storage.k8s.io/snapshotter-secret-namespace: openshift-storage`
  
  IMO this is good to merge now

**Comment by BlaineEXE** [comment]
LGTM, but I'll let @nixpanic approve since he had some prior comments.

**Review by nixpanic — APPROVED** [review]

**Review by agarwal-mudit — APPROVED** [review]

**Pull request merged**
